pkgs/build-support/compress-drv/default.nix

```
@@ -0,0 +1,63 @@
+/*
+  compressDrv compresses files in a given derivation.
+
+  Inputs:
+
+  - formats :: [String]
+
+      List of file extensions to compress.
+
+      Example: ["txt" "svg" "xml"]
+
+  - compressors :: {String -> String}
+
+      Map a desired extension (e.g. `gz`) to a compress program.
+
+      The compressor program that will be executed to get the `COMPRESSOR`
+      extension. The program should have a single " {}", which will be the
+      replaced with the target filename.
+
+      Compressor must:
+      - read symlinks (thus --force is needed to gzip, zstd, xz).
+      - keep the original file in place (--keep).
+
+      Example:
+
+        {
+          xz = "${xz}/bin/xz --force --keep {}";
+        }
+
+   See compressDrvWeb, which is a wrapper on top of compressDrv, for broader
+   use examples.
+*/
+{
+  lib,
+  xorg,
+  runCommand,
+}:
+drv:
```

The placement is a little off,  

```suggestion
{
  lib,
  xorg,
  runCommand,
}
/** 
Your documentation
*/
drv:
```

You can find the correct placement, by typing this into your repl.
Nix will print the source position of the lambda, you want to document

```nix repl
nix-repl> pkgs.compressDrv.__functor 1 
«lambda @ nixpkgs/pkgs/build-support/compress-drv/default.nix:38:1»
```